libs/utils/analysis/frequency_analysis.py

```diff
@@ -163,6 +163,124 @@ def _dfg_cpu_frequency_transition_rate(self, cpu):
 # Plotting Methods
 ###############################################################################
 
+    def plotPeripheralClock(self, clk, title='Peripheral Frequency'):
+        """
+        Produce graph plotting the frequency of a particular peripheral clock
+
+        :param title: The title for the chart
+        :type  title: str
+
+        :param clk: The clk name to chart
+        :type  clk: str
+
+        :raises: KeyError
+        """
+        if not self._trace.hasEvents('clock_set_rate'):
+            self._log.warning('Events [clock_set_rate] not found, plot DISABLED!')
+            return
+        rate_df = self._dfg_trace_event('clock_set_rate')
+        enable_df = self._dfg_trace_event('clock_enable')
+        disable_df = self._dfg_trace_event('clock_disable')
+
+        rate_df = rate_df[rate_df.clk_name == clk]
+        rate_df['clock_setting'] = rate_df['rate']/1e6
```

Comment:
I get setting with copy warnings here. If this is needed for all clock-related analysis, this could be added to trace sanitization (in **libs/utils/trace.py**) and applied to all the traces (no filtering on clk).

Otherwise, if this is just needed for this analysis, you could do this:
```
rate_df = rate_df[rate_df.clk_name == clk].copy()
```

Reply:
Adopted changes, although I couldn't reproduce the warning here with my setup (pandas 0.22)